### ISSUE TITLE
ClientConfigurationError: Scopes are required to obtain an access token

### ISSUE THREAD

**Issue by bitflower** [microsoftgraph/msgraph-sdk-javascript#193]
# Bug Report

## Prerequisites

-   [x] Can you reproduce the problem?
-   [x] Are you running the latest version?
-   [x] Are you reporting to the correct repository?
-   [x] Did you perform a cursory search?

## Description

I'm getting the given error although setting the `scopes`:
```
const graphScopes = ['User.Read', 'Mail.Send']; 
getAuthProvider(msalInstance, graphScopes);

...

import * as MSALAuthenticationProvider from '@microsoft/microsoft-graph-client/lib/es/browser/MSALAuthenticationProvider';
export const getAuthProvider = (userAgentApplication, scopes) => {
  console.log('getAuthProvider');
  authProvider = new MSALAuthenticationProvider.MSALAuthenticationProvider(
    userAgentApplication,
    scopes
  );
};

...

import * as EsClient from '@microsoft/microsoft-graph-client/lib/es/Client';
export const getGraphCLient = (): EsClient.Client => {
  console.log('getGraphCLient');

  const options: ClientOptions = {
    authProvider
  };
  //   const Client = Client.Client;
  graphClient = EsClient.Client.initWithMiddleware(options);

  return graphClient;
};

...

    try {
      let userDetails = await graphClient.api('/me').get();
      console.log(userDetails);
    } catch (error) {
      throw error;
    }
```

**Console Errors:** [Is there any console error]
```
app-root.entry.js:7685 Uncaught (in promise) 
GraphError {statusCode: -1, code: "Error", message: "ClientConfigurationError: Scopes are required to obtain an access token. Given value: undefined", requestId: null, date: Tue Jun 18 2019 11:06:15 GMT+0200 (Mitteleuropäische Sommerzeit), …}
body: "Error: ClientConfigurationError: Scopes are required to obtain an access token. Given value: undefined"
code: "Error"
date: Tue Jun 18 2019 11:06:15 GMT+0200 (Mitteleuropäische Sommerzeit) {}
message: "ClientConfigurationError: Scopes are required to obtain an access token. Given value: undefined"
requestId: null
statusCode: -1
```

**Screenshots:** [If applicable, add screenshots to help explain your problem]
<img width="1344" alt="Bildschirmfoto 2019-06-18 um 11 10 55" src="https://user-images.githubusercontent.com/6666636/59669083-dc268500-91b9-11e9-8a15-9a8998ad1e71.png">

**Expected behavior:** [What you expected to happen]
Can use the graph client as expected.

## Repo to re-produce
https://github.com/bitflower/stencil-ms-graph

## Additional context
This is inside a`stencil` app bundled up with `rollup`.

**Comment by bitflower** [comment]
My Typescript imports look a bit weird I know. Can't imagine this being the root cause though?

**Comment by bitflower** [comment]
<img width="1263" alt="Bildschirmfoto 2019-06-18 um 11 25 32" src="https://user-images.githubusercontent.com/6666636/59670248-e9dd0a00-91bb-11e9-887d-eb416f1df07d.png">

What I get back also looks fine.

**Comment by bitflower** [comment]
Made a couple of logs and found out that the instance of UserAgentApplication in the client seems to be a different one than the one used in the first place:

```
getMsalInstance
app-root.entry.js:5786 Msal.UserAgentApplication UserAgentApplication {authResponseCallback: ƒ, tokenReceivedCallback: null, errorReceivedCallback: null, config: {…}, redirectCallbacksSet: true, …}
app-root.entry.js:8491 handleLogin
app-root.entry.js:5791 getLogin
app-root.entry.js:3231 UserAgentApplication.prototype.loginPopup ["user.read"]
app-root.entry.js:5191 UserAgentApplication.prototype.validateInputScope ["user.read"] false
app-root.entry.js:5204 UserAgentApplication.prototype.validateInputScope ["user.read"]
app-root.entry.js:5209 UserAgentApplication.prototype.validateInputScope 1
app-root.entry.js:1730 ServerRequestParameters.prototype.parseScope (3) ["user.read", "openid", "profile"]
app-root.entry.js:5886 token {uniqueId: "00000000-0000-0000-41f5-edfbb62f164a", tenantId: "9188040d-6c67-4c5b-b112-36a304b66dad", tokenType: "id_token", idToken: IdToken, accessToken: null, …}
app-root.entry.js:6032 getAuthProvider
app-root.entry.js:8495 AuthProvider MSALAuthenticationProvider {scopes: Array(3), userAgentApplication: UserAgentApplication}
app-root.entry.js:8472 getGraphCLient
app-root.entry.js:8497 graphClient Client {config: {…}, httpClient: HTTPClient}
app-root.entry.js:8499 Reading user
app-root.entry.js:6746 https://graph.microsoft.com/v1.0/me
app-root.entry.js:3709 UserAgentApplication.prototype.acquireTokenSilent undefined
app-root.entry.js:5191 UserAgentApplication.prototype.validateInputScope undefined true
app-root.entry.js:3231 UserAgentApplication.prototype.loginPopup undefined
app-root.entry.js:5191 UserAgentApplication.prototype.validateInputScope undefined false
app-root.entry.js:1730 ServerRequestParameters.prototype.parseScope (2) ["openid", "profile"]
app-root.entry.js:3709 UserAgentApplication.prototype.acquireTokenSilent undefined
app-root.entry.js:5191 UserAgentApplication.prototype.validateInputScope undefined true
app-root.entry.js:3411 UserAgentApplication.prototype.acquireTokenPopup undefined
app-root.entry.js:5191 UserAgentApplication.prototype.validateInputScope undefined true
```

You can see the `scopes` being empty from line `Reading user` on which is the moment the msgraph client is making its call.

**Comment by bitflower** [comment]
I think in this line we should handover more than just `scopes`? It's supposed to be a `request` object with `scopes being nested inside`
https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/d571846b6c5a62a9dfde88d915904052c7190ecd/src/browser/MSALAuthenticationProvider.ts#L80

**Comment by bitflower** [comment]
The second issue ist that in this line the returned `accessToken` *containing* the `accesstoken` so we shoulds return `accesstoken.accessToken`:
https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/d571846b6c5a62a9dfde88d915904052c7190ecd/src/browser/MSALAuthenticationProvider.ts#L81

**Comment by darrelmiller** [comment]
@bitflower  That last issue looks like a mismatch between the MSAL.js version and the AuthenticationProvider.  There were breaking changes in the 1.0 release of MSAL.js that changed the return value from simply and authToken to an object containing an auth token.  Can you confirm which version of msal.js and GraphSDK that you are using?  /cc @muthurathinam

**Comment by bitflower** [comment]
Hey, thanks for the fast reply. Sure:
```
    "@microsoft/microsoft-graph-client": "^1.7.0",
    "msal": "^1.0.1",
```

**Comment by darrelmiller** [comment]
@bitflower I believe the fix is in our 2.0 release here https://github.com/microsoftgraph/msgraph-sdk-javascript/pull/179  We are just waiting for this to be merged.

**Comment by bitflower** [comment]
Hi @darrelmiller, that‘s good to know. I‘ll follow the other issue! Thanks!

**Comment by muthurathinam** [comment]
@bitflower , Darrel is right. Library version is compatible with MSAL v0.2.4 and the support for v1.x.x will be out with 2.0 release.

**Comment by bitflower** [comment]
Thanks for the update!

**Comment by muthurathinam** [comment]
We have released our preview version for 2.0.0 which fixes the issue. To install the preview version of the library, use `npm install @microsoft/microsoft-graph-client@Preview`.
Refer [this release notes](https://github.com/microsoftgraph/msgraph-sdk-javascript/releases/tag/2.0.0-Preview.1) for the list of changes.

**Comment by mlynam** [comment]
Thanks guys!
